core/templates/pages/learner-dashboard-page/new-learner-dashboard/community-lessons-tab.component.ts

```diff
@@ -0,0 +1,309 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for community lessons tab in the Learner Dashboard
+ * page.
+ */
+
+import { Component, Input } from '@angular/core';
+import { LearnerDashboardActivityBackendApiService } from 'domain/learner_dashboard/learner-dashboard-activity-backend-api.service';
+import { LearnerDashboardPageConstants } from '../learner-dashboard-page.constants';
+import { LearnerExplorationSummary } from 'domain/summary/learner-exploration-summary.model';
+import { CollectionSummary } from 'domain/collection/collection-summary.model';
+import { ProfileSummary } from 'domain/user/profile-summary.model';
+import { DeviceInfoService } from 'services/contextual/device-info.service';
+
+ @Component({
+   selector: 'oppia-community-lessons-tab',
+   templateUrl: './community-lessons-tab.component.html'
+ })
+export class CommunityLessonsTabComponent {
+  constructor(
+    private learnerDashboardActivityBackendApiService: (
+      LearnerDashboardActivityBackendApiService),
+    private deviceInfoService: DeviceInfoService) {
+  }
+  @Input() incompleteExplorationsList: LearnerExplorationSummary[];
+  @Input() incompleteCollectionsList: CollectionSummary[];
+  @Input() completedExplorationsList: LearnerExplorationSummary[];
+  @Input() completedCollectionsList: CollectionSummary[];
+  @Input() explorationPlaylist: LearnerExplorationSummary[];
+  @Input() collectionPlaylist: CollectionSummary[];
+  @Input() subscriptionsList: ProfileSummary[];
+  @Input() completedToIncompleteCollections: string[];
+  noCommunityLessonActivity: boolean;
+  noPlaylistActivity: boolean;
+  totalIncompleteLessonsList: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  totalCompletedLessonsList: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  totalLessonsInPlaylist: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  allCommunityLessons: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  displayIncompleteLessonsList: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  displayCompletedLessonsList: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  displayLessonsInPlaylist: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  displayInCommunityLessons: (
+    LearnerExplorationSummary | CollectionSummary)[] = [];
+  selectedSection: string;
+  dropdownEnabled: boolean;
+  showMore = {
```

Comment:
Can this be more descriptive? Something like `showMoreInSection` or `expandSection` or similar?

Reply:
Done.